api/keystore/service.go

```diff
@@ -92,10 +92,28 @@ func (ks *Keystore) Initialize(log logging.Logger, db database.Database) {
 	ks.bcDB = prefixdb.New([]byte("bcs"), db)
 }
 
+var mapping jsoncodec.RPCRestMap = jsoncodec.RPCRestMap{
+	"/api/keystore/user/create": map[string]string{
+		"post": "keystore.CreateUser",
+	},
+	"/api/keystore/user/delete": map[string]string{
+		"post": "keystore.DeleteUser",
+	},
+	"/api/keystore/user/list": map[string]string{
+		"post": "keystore.ListUsers",
+	},
+	"/api/keystore/user/import": map[string]string{
+		"post": "keystore.ImportUser",
+	},
+	"/api/keystore/user/export": map[string]string{
+		"post": "keystore.ExportUser",
+	},
+}
+
 // CreateHandler returns a new service object that can send requests to thisAPI.
 func (ks *Keystore) CreateHandler() *common.HTTPHandler {
 	newServer := rpc.NewServer()
-	codec := jsoncodec.NewCodec()
+	codec := jsoncodec.RestCodec{Mapping: mapping}
```

Comment:
I'm not sure this should be implemented as a codec. Codecs in Gorilla are there to serialize/deserialize different content types into single interface parameters to feed the processing method. This is not what we want to do here.

Also in this case you replace the existing `NewCodec` which seems to have a real purpose (convert the first character of the method to upper case), just replacing it like that will likely break current behavior.

Reply:
NewCodec is there. In the `rest_codec.go` i defined below, I added that if path is prefixed with /api use the ret codec . Else sent the /ext request to newcodec. I kept compatibility in mind, keeping it 100% compatible.

With codec for gorilla, i am serializing  data to goeilla/rpc fields. As you said, the  json/codec.go uses the same approach converting method name to upper case. 

This approach saves to create entirely new Server and prevents duplication of code. Apart from that, I kept it very easy to integrate with current code base of gecko.

Reply:
This approach seems pretty elegant to me. @glemercier are there consequences to this that I am not seeing right now? Can you help me understand?  You may very well have a damn good point I just don't see it yet. Can you two take it to Discord with me? I'm curious to hear your feedback, and I will be denying this PR as it's incomplete and I want a complete PR for Stephen and Dan to review.